agent/exploits/cve_2019_16278.py

```diff
@@ -0,0 +1,101 @@
+"""Agent Asteroid implementation for CVE-2019-16278"""
+
+import datetime
+import http.client as http_client
+from urllib import parse as urlparse
+
+import requests
+from requests import exceptions as requests_exceptions
+
+from agent import definitions
+from agent import exploits_registry
+from agent.exploits import webexploit
+
+VULNERABILITY_TITLE = "NOSTROMO NHTTPD DIRECTORY TRAVERSAL VULNERABILITY"
+VULNERABILITY_REFERENCE = "CVE-2019-16278"
+VULNERABILITY_DESCRIPTION = """Directory Traversal in the function http_verify in nostromo nhttpd through 1.9.6 allows an attacker to achieve remote code execution via a crafted HTTP request.
+"""
+RISK_RATING = "CRITICAL"
+
+DEFAULT_TIMEOUT = datetime.timedelta(seconds=90)
+
+
+@exploits_registry.register
+class CVE201916278Exploit(webexploit.WebExploit):
+    accept_request = definitions.Request(method="GET", path="/")
+    metadata = definitions.VulnerabilityMetadata(
+        title=VULNERABILITY_TITLE,
+        description=VULNERABILITY_DESCRIPTION,
+        reference=VULNERABILITY_REFERENCE,
+        risk_rating=RISK_RATING,
+    )
+
+    def accept(self, target: definitions.Target) -> bool:
+        """Rule: heuristically detect if a specific target is valid.
+
+        Args:
+            target: Target to verify
+
+        Returns:
+            True if the target is valid; otherwise False.
+        """
+        target_endpoint = urlparse.urljoin(target.origin, self.accept_request.path)
+        try:
+            req = requests.Request(
+                method=self.accept_request.method,
+                url=target_endpoint,
+                data=self.accept_request.data,
+            ).prepare()
+            resp = self.session.send(req, timeout=DEFAULT_TIMEOUT.seconds)
+        except requests_exceptions.RequestException:
+            return False
+
+        try:
+            server_header = resp.headers.get("Server", "")
+            if "nostromo" in server_header:
+                return True
+        except Exception:
```

Comment:
Is this intentional?

Reply:
what do you mean?

Reply:
Catching the generic exception.

Reply:
why do you need to catch in the first place ? in the try block i cant see anything that can raise exception

Reply:
yes true! it was for other checks that are removed!